client/src/telethon/_impl/client/events/queries.py

```diff
@@ -43,6 +51,71 @@ def data(self) -> bytes:
         assert self._raw.data is not None
         return self._raw.data
 
+    @property
+    def via_inline(self) -> bool:
+        """
+        Whether the button was clicked in an inline message.
+
+        If it was, most likely bot is not in chat, and the :meth:`chat` property will return :data:`None`,
+        same for :meth:`get_message` method, however editing the message, using :meth:`message_id` property
+        and :meth:`answer` method will work.
+        """
+        return isinstance(self._raw, types.UpdateInlineBotCallbackQuery)
+
+    @property
+    def message_id(self) -> typing.Union[int, abcs.InputBotInlineMessageId]:
+        """
+        The ID of the message containing the button that was clicked.
+        
+        If the message is inline, :class:`abcs.InputBotInlineMessageId` will be returned.
+        You can use it in :meth:`~telethon._tl.functions.messages.edit_inline_bot_message` to edit the message.
+
+        Else, usual message ID will be returned.
+        """
+        return self._raw.msg_id
+
+    @property
+    def chat(self) -> Optional[Chat]:
+        """
+        The :term:`chat` where the button was clicked.
+
+        This will be :data:`None` if the message with the button was sent from a user's inline query, except in channel.
+        """
+        if isinstance(self._raw, types.UpdateInlineBotCallbackQuery):
+            owner_id = None
+            if isinstance(self._raw.msg_id, types.InputBotInlineMessageId):
+                _, owner_id = struct.unpack("<ii", struct.pack("<q", self._raw.msg_id.id))
+            elif isinstance(self._raw.msg_id, types.InputBotInlineMessageId64):
+                _ = self._raw.msg_id.id
+                owner_id = self._raw.msg_id.owner_id
+
+            if owner_id is None:
+                return None
+
+            if owner_id > 0:
+                # We can't know if it's really a chat with user, or an ID of the user who issued the inline query.
+                # So it's better to return None, then to return wrong chat.
```

Comment:
Typo `then` -> `than`.

Still a bit weird that there's no way to tell. Does the update type not contain any other hint elsewhere?